lib/archethic/shared_secrets/node_renewal.ex

```diff
@@ -114,6 +114,35 @@ defmodule ArchEthic.SharedSecrets.NodeRenewal do
     )
   end
 
+  @spec new_origin_shared_secrets_transaction(
+          origin_public_keys :: list(Crypto.key()),
+          secret_key :: binary()
+        ) :: Transaction.t()
+  def new_origin_shared_secrets_transaction(
+        origin_public_keys \\ [<<0::0, 0::0, :crypto.strong_rand_bytes(32)::binary>>],
+        secret_key \\ :crypto.strong_rand_bytes(32)
+      )
+      when is_binary(secret_key) and
+             is_list(origin_public_keys) do
+    {new_shared_origin_pub_key, new_shared_origin_priv_key} = Crypto.generate_random_keypair()
+    origin_public_key = Enum.at(origin_public_keys, 0)
+
+    encrypted_origin_shared_priv_key =
+      Crypto.ec_encrypt(new_shared_origin_priv_key, origin_public_key)
+
+    secret = encrypted_origin_shared_priv_key
+
+    Transaction.new(
+      :origin_shared_secrets,
+      %TransactionData{
+        content: new_shared_origin_pub_key,
```

Comment:
I don't mind changing it, but we need to be consistent with `SharedSecrets.MemTablesLoader`

Reply:
As we are adding public key, content is want we expect to have i.e binary with pub_keys concatenated. Just once confirm this is exactly want we want then can add more pub_keys by concating

Reply:
From the YP perspective, it was mentioned a list of public keys but per chain of family.
Now the question will be to have a single chain or multiple chain, but otherwise I think you can concatenate generated origin public keys in the transaction's content

Reply:
But you would have to adapt the mem_table_loader then